_posts/2022-04-27-newpipe-0.23.0.md

```diff
@@ -0,0 +1,176 @@
+
+- As a follow-up, [@TiA4f8R](https://github.com/TiA4f8R) also ensured that subtitles are extracted for age-restricted videos.
+
+- The infamous 'n' parameter used by YouTube changed again, which caused throttling. That was fixed (again) by [@Stypox](https://github.com/Stypox).
+
+- If you used external buttons (like on a Bluetooth headphone or speaker) to resume playback, the player controls UI would remain on screen forever, instead of fading out after a few moments. [@ktprograms](https://github.com/ktprograms) fixed this.
+
+- The update check would randomly fail while the app was in the background. [@TacoTheDank](https://github.com/TacoTheDank) updated some deprecated code to fix this.
+
+- The new error notifications added by [@Stypox](https://github.com/Stypox) in a previous release crashed the app on KitKat, which they have now fixed.
+
+- [@karyogamy](https://github.com/karyogamy) also fixed playback being reset to the beginning when the player type was changed (e.g. popup -> audio or audio -> main).
+
+- [@mauriciocolli](https://github.com/mauriciocolli) fixed the layout of the video resolution selection menu in the download dialogue, which earlier left space for a non-existent mute icon to the left of each option.
+
+- If you tried to rotate the screen while the "Add to playlist" dialogue was open, the app would crash, which has been fixed by [@GGAutomaton](https://github.com/GGAutomaton). Thanks!
+
+## Nerd Talk
+
+- dependabot updates:
+  - Bump actions/checkout from v2 to v3
+  - Bump spotbugs-annotations from 4.5.3 to 4.6.0
+  - Bump gson from 2.8.9 to 2.9.0
+  - Bump actions/cache from 2 to 3
+  - Bump rhino from 1.7.13 to 1.7.14
+
+- [@litetex](https://github.com/litetex) refactored the `YoutubeCommentsExtractor` class to use the Java 8 Stream API, which greatly simplifies the code and reduces potential bugs.
+
+- [@FireMasterK](https://github.com/FireMasterK) added support for extracting the channel subscriber count in the `StreamInfo` class.
+
+- [@FireMasterK](https://github.com/FireMasterK) fixed a PeerTube test, and also removed the need for a new `CommentsInfo` instance each time one of the methods of this class had to be called, reducing the RAM burden caused by the unneeded instances.
+
+- [@litetex](https://github.com/litetex) realised that our hardcoded SoundCloud ClientID was quickly after each release out of date, so there was no point in updating it again and again. It has been removed completely. Now we don't have to chase after a moving goalpost of an optimisation, which hardly benefited users. Thanks, litetex!
+
+- [@Stypox](https://github.com/Stypox) added the `MockOnly` JUnit 5 extension to all the YouTube extraction tests so that they work for Mixes. Moreover, the `DownloaderFactory` class's methods were made static, shaving off a little bit of CPU overhead each time that code is run.
```

Comment:
```suggestion
- [@Stypox](https://github.com/Stypox) added the `MockOnly` JUnit 5 extension so that a test can be instructed to only run when mocking the downloader. This was needed for YouTube Mixes. Moreover, the `DownloaderFactory` class's methods were made static, shaving off a little bit of CPU overhead each time that code is run.
```